_posts/en/newsletters/2019-05-29-newsletter.md

```diff
@@ -0,0 +1,631 @@
+(or some other protocol address for delivery of the committed outputs).
+
+### More advanced usage
+
+The transaction size (and thus fee cost) of using COSHV in the previous
+subsection's example is basically the same as using the pre-signed
+transaction format with current Bitcoin transactions.  However, the
+ability to use Taproot to allow the receivers to choose from multiple
+spending conditions adds a significant amount of flexibility to the approach.
+
+In our previous examples, getting the distribution transaction confirmed
+was an all-or-nothing endeavor.  Either all ten receivers finally
+received their money at the same time, or all of them were still waiting
+for that confirmation to occur.  However, with Taproot and COSHV,
+it's possible to allow each receiver to individually withdraw their
+money from the covenant while also allowing most the other receivers to
+keep their money offchain for the moment.
+
+For example, let's imagine Alice paid each of our ten receivers 1 BTC
+using a tree of outputs where each output paid two other outputs until
+an individual's money was withdrawn from the covenant.  Or, at each
+level, an alternative Taproot option would allow paying out all
+remaining outputs at that point.  A complete tree might for ten
+participants might look like this:
+
+![Example output tree for ten participants](/img/posts/2019-05-tx-tree-5.png)
+{:.center}
+
+If receiver Bob wants to individually withdraw now, he could naively
+follow the whole path until he received his output, allowing him to
+create only five outputs instead of the ten he'd need to create if
+everyone had to exit at the same time.
+
+![Example output tree to allow participant Bob to exit](/img/posts/2019-05-tx-tree-6.png)
+{:.center}
+
+Since Bob is the first participant to withdraw, he has to create more
+outputs than everyone else, about `log2(number_of_participants)`.
+Anyone else exiting after Bob would need to create even fewer outputs as
+they could just start spending from one of the intermediate COSHV
+outputs Bob had created.  Some people (like Charlie in the above
+example) would get their exit outputs as a byproduct of another person's
+exit.  Overall, the total number of outputs that need to be created
+would be about double the number of participants.  However, this could
+be just the one output per participant from the previous subsection if
+one of the taproot leaves at the top level also allowed just sending the
+regular batched payment transaction---allowing everyone to exit together
+efficiently.
+
+This balanced tree construction provides a good generic starting part for
+allowing individuals to exit from the covenant without having to pay all
+the exit costs themselves.  When applying COSHV to specific cases where
+it's known some exits are more likely than others, it's possible to
+use unbalanced trees to make the more likely outcomes less expensive.
+We expect that is COSHV is adopted and sees regular use, tools such
```

Comment:
s/is COSHV is/if COSHV is/

Reply:
also consider changing this to say "if COSHV becomes part of the Bitcoin protocol and sees regular use" or similar to emphasise that this is still a proposal